workflow/tools/util_in_docker/generate_song_payload.py

```diff
@@ -212,6 +226,42 @@ def retrieve_metrics_file(metrics_directory, read_group):
             return file
     raise Exception("The metrics file %s does not exist." % (file_full_path))
 
+def get_workflow_data(wf_name, wf_version, execution_runner_name, execution_runner_version, execution_job_id,yaml_data):
+    return {
+        'name': wf_name,
+        'version': wf_version,
+        'execution': {
+            'runner_name': execution_runner_name,
+            'runner_version': execution_runner_version,
+            'job_id': execution_job_id
+        },
+        'input': get_workflow_data_files(yaml_data)
+    }
+
+def get_workflow_data_files(yaml_data):
+    files = []
+    for file in yaml_data.get('files'):
```

Comment:
same file can be duplicated in the YAML, need to deduplicate here.

Reply:
Which field can be duplicated? Would it be all of the 4 fields?
repository, bundle_id, object_id, file_name

Reply:
here is an example: https://github.com/icgc-dcc/pcawg2-meta/blob/master/tests/preprocessing/50Candidate-PCAWG2-donors_20190215/WGS/OV-AU/AOCS-122/AOCS-122-1-4.tumour/metadata.yaml.

Search for `7aba9867-ffba-5e57-96d0-300c80839504`, same file occurred four times in the YAML